listen.go

```diff
@@ -564,6 +584,12 @@ func (ln *listener) reader(ctx context.Context) {
 				break
 			}
 
+			if p.Header().Addr.String() != conn.RemoteAddr().String() {
```

Comment:
Could you add a flag to the config so we can turn this on or off? I'm a little worried we'll run into issues w/ this check & our combination of LBs & Envoy so it'd be nice to have an easy way to disable it

Reply:
Yeah that's fair. I'd still hope the ports are distinct though

Reply:
Done. 

Another thing I don't love about this, I thought UDP is supposed to be nice because it's connectionless, a client can change ip address, while changing networks for example, and things still work. Which would seem to be nice if streaming from a mobile device and you go from wifi to cellular or something.

But nope, I think SRT recommends blocking changing IPs anyway? https://haivision.github.io/srt-rfc/draft-sharabayko-srt.html#section-8-3.4.1
> An eavesdropper can hijack existing connections only if it steals the IP and port of one of the parties. If some stream addresses an existing SRT receiver by its SRT socket ID, IP, and port number, but arrives from a different IP or port, the SRT receiver ignores it.

Reply:
Yeah I don't love that either but I think we'd run into other issues if IPs change (would potentially get hashed to a different liveingest pod)